app/models/concerns/smooch_search.rb

```diff
@@ -4,23 +4,34 @@ module SmoochSearch
   extend ActiveSupport::Concern
 
   module ClassMethods
+
     # This method runs in background
     def search(app_id, uid, language, message, team_id, workflow, provider = nil)
       platform = self.get_platform_from_message(message)
       begin
         sm = CheckStateMachine.new(uid)
         self.get_installation(self.installation_setting_id_keys, app_id) if self.config.blank?
         RequestStore.store[:smooch_bot_provider] = provider unless provider.blank?
-        results = self.get_search_results(uid, message, team_id, language).select do |pm|
-          pm = Relationship.confirmed_parent(pm)
-          report = pm.get_dynamic_annotation('report_design')
-          !report.nil? && !!report.should_send_report_in_this_language?(language)
-        end.collect{ |pm| Relationship.confirmed_parent(pm) }.uniq
-        if results.empty?
+        query = self.get_search_query(uid, message)
+        results = self.get_search_results(uid, query, team_id, language).collect{ |pm| Relationship.confirmed_parent(pm) }.uniq
+        reports = results.collect{ |pm| pm.get_dynamic_annotation('report_design') }.reject{ |r| r.nil? }.collect{ |r| r.report_design_to_tipline_search_result }.select{ |r| r.should_send_in_language?(language) }
+
+        # Extract explainers from matched media if they don't have published fact-checks but they have explainers
+        reports = results.collect{ |pm| pm.explainers.to_a }.flatten.uniq.first(3).map(&:as_tipline_search_result) if !results.empty? && reports.empty?
+
+        # Search for explainers if fact-checks were not found
+        if results.empty? && query['type'] == 'text'
```

Comment:
Should this be results.empty or reports.empty?

Reply:
Good catch, thanks, this should be `reports.empty?`.